core/src/main/scala/com/metamx/tranquility/druid/IndexService.scala

```diff
@@ -87,7 +92,24 @@ class IndexService(
     val statusRequest = HttpGet("/druid/indexer/v1/task/%s/status" format taskId)
     call(statusRequest) map {
       d =>
-        d.get("status") map (sd => IndexStatus.fromString(str(dict(sd)("status")))) getOrElse TaskNotFound
+        dict(d).get("status") map (sd => IndexStatus.fromString(str(dict(sd)("status")))) getOrElse TaskNotFound
+    }
+  }
+
+  def runningTasks(): Future[Map[TaskId, TaskHostPort]] = {
+    val request = HttpGet("/druid/indexer/v1/runningTasks")
```

Comment:
Is there a way to add a timeout to this?

For example, if an overlord locks up and stops returning results (just holding connection), what is the impact here?